src/sonic-config-engine/minigraph.py

```diff
@@ -81,7 +82,72 @@ def parse_device(device):
             deployment_id = node.text
     return (lo_prefix, mgmt_prefix, name, hwsku, d_type, deployment_id)
 
-def parse_png(png, hname):
+def formulate_ecmp_entry(dpg_ecmp_content, port_device_map):
+    banks_enumerated = {}
+    lcm_array = []
+    LCM = 0
+    FG_NHG_MEMBER = [{}, {}]
+    FG_NHG_PREFIX = [{}, {}]
+    FG_NHG = [{}, {}]
+    FG_NEIGH = [{}, {}]
+    COMP_FG_NHG_MEMBER = {}
+    COMP_FG_NHG_PREFIX = {}
+    COMP_FG_NHG = {}
+    COMP_FG_NEIGH = {}
+    neigh_key_ipv4 = []
+    neigh_key_ipv6 = []
+    ipv4_tag = "fgnhg_v4"
+    ipv6_tag = "fgnhg_v6"
+    port_nhipv4_map = dpg_ecmp_content[0]
+    port_nhipv6_map = dpg_ecmp_content[1]
+    nhgaddr = dpg_ecmp_content[2]
+    nhgvlan = dpg_ecmp_content[3]
+    nhipv4_device_map = {port_nhipv4_map[x]: port_device_map[x] for x in port_device_map
+                         if x in port_nhipv4_map}
+    nhipv6_device_map = {port_nhipv6_map[x]: port_device_map[x] for x in port_device_map
+                         if x in port_nhipv6_map}
+    nhdevices = sorted(list(set(nhipv4_device_map.values())))
+    nhdevices_bank_map = {device: bank for bank, device in enumerate(nhdevices)}
+    nhipv4_bank_map = {ip: nhdevices_bank_map[device] for ip, device in nhipv4_device_map.items()}
+    nhipv6_bank_map = {ip: nhdevices_bank_map[device] for ip, device in nhipv6_device_map.items()}
+    for value in nhdevices_bank_map.values():
```

Comment:
IPv4 and IPv6 could have a different enumeration of banks and as a result a different lcm. I would also suggest moving this calculation of lcm to a helper function.

Reply:
ok will do this